articles/techniques/type-model.md

```diff
@@ -109,8 +108,11 @@ The types before and after each arrow can be whatever type we wish, including ot
 For instance, we can pass the function `SquareOperation` defined above to any input expecting type ` ((Qubit => Unit) -> (Qubit => Unit))`.
 Informally, we can read that type as "a classical function which takes operations on a single qubit and returns operations on a single qubit."
 
-In order to use the `Controlled` and `Adjoint` variants of an operation type, we need to indicate that the values of that type support the variants we wish to call.
-This is done by adding constraints to the operation type, as in `(Qubit => Unit : Adjoint)`, which indicates an adjointable operation acting on one qubit to produce an empty tuple as its output.
+In order to require support for the `Controlled` and/or `Adjoint` functor in an operation type, we need to add an annotation indicating the corresponding characteristics.
+An annotation `is Ctl` for example indicates that the operation is controllable. 
+If we want to require that an operation of that type supports both the `Adjoint` and `Controlled` functor we can express this as `(Qubit => Unit is Adj + Ctl)`. 
+The used operation characteristics `Adj` and `Ctl` strictly speaking are two pre-defined sets of labels, where each label indicates a particular operation characteristics like e.g. support for a particular functor.
+Hence, `+` is used to indicate the union of those two sets, and `*` is used to indicate the intersection - i.e. the labels that are common to both sets.  
```

Comment:
Actually, we use `+` as if it were intersection: when we say that an operation `is Adj + Ctl`, we mean that it has both adjoint and controlled specializations, which is to say it's in the intersection of `Adj` and `Ctl`...

Reply:
While I would have like to have Adj be the set of all operations that is adjointable, we unfortunately can't do that, precisely because no inverse element exists for intersection. Hence, Adj is one of the predefined *sets of labels*, specifically Adj = {"Adjointable"}, and Ctl = {"Controllable"}. Adj + Ctl is indeed the set union Adj + Ctl = {"Adjointable", "Controllable"}, used to indicate that the labels "Adjointable" and "Controllable" are both true for that operation.